src/SW_netCDF.c

```diff
@@ -0,0 +1,360 @@
+ * @param[in] domFileName Name of the domain netCDF file
+ * @param[in,out] LogInfo Holds information dealing with logfile output
+*/
+void SW_NC_read_domain(SW_DOMAIN* SW_Domain, const char* domFileName,
+                       LOG_INFO* LogInfo) {
+
+    (void) SW_Domain;
+    (void) domFileName;
+    (void) LogInfo;
+}
+
+/**
+ * @brief Check that content is consistent with domain
+ *
+ * @param[in] SW_Domain Struct of type SW_DOMAIN holding constant
+ *  temporal/spatial information for a set of simulation runs
+ * @param[in] fileName Name of netCDF file to test
+ * @param[in,out] LogInfo Holds information dealing with logfile output
+*/
+void SW_NC_check(SW_DOMAIN* SW_Domain, const char* fileName,
+                 LOG_INFO* LogInfo) {
+
+    (void) SW_Domain;
+    (void) fileName;
+    (void) LogInfo;
+}
+
+/**
+ * @brief Create domain netCDF template if it does not already exists
+ *
+ * @param[in] SW_Domain Struct of type SW_DOMAIN holding constant
+ *  temporal/spatial information for a set of simulation runs
+ * @param[in,out] LogInfo Holds information dealing with logfile output
+*/
+void SW_NC_create_domain_template(SW_DOMAIN* SW_Domain, LOG_INFO* LogInfo) {
+
+    int* domFileID = &SW_Domain->PathInfo.domainFileID;
+    int sDimID = 0, xDimID = 0, yDimID = 0; // varID is not used
+    int domDims[2]; // Either [yDimID, xDimID] or [sDimID, 0]
+    int nDomainDims;
+
+    if(FileExists(DOMAIN_TEMP)) {
+        LogError(LogInfo, LOGERROR, "Could not create new domain template. "
+                                    "This is due to the fact that it already "
+                                    " exists. Please modify it and change the name.");
+        return; // Exit prematurely due to error
+    }
+
+    if(nc_create(DOMAIN_TEMP, NC_NETCDF4, domFileID) != NC_NOERR) {
+        LogError(LogInfo, LOGERROR, "Could not create new domain template due "
+                                    "to something internal.");
+        return; // Exit prematurely due to error
+    }
+
+    if(strcmp(SW_Domain->DomainType, "s") == 0) {
+        nDomainDims = 1;
+
+        // Create s dimension/domain variables
+        fill_domain_netCDF_s(SW_Domain->nDimS, domFileID, &sDimID,
+                             &xDimID, &yDimID, LogInfo);
+        if(LogInfo->stopRun) {
+            nc_close(*domFileID);
+            return; // Exit prematurely due to error
+        }
+
+        domDims[0] = sDimID;
+        domDims[1] = 0;
+    } else {
+        nDomainDims = 2;
+
+        fill_domain_netCDF_xy(SW_Domain->nDimX, SW_Domain->nDimY, domFileID,
+                              &xDimID, &yDimID, LogInfo);
+        if(LogInfo->stopRun) {
+            nc_close(*domFileID);
+            return; // Exit prematurely due to error
+        }
+
+        domDims[0] = yDimID;
+        domDims[1] = xDimID;
+    }
+
+    // Create domain variable
+    create_netCDF_var("domain", domDims, domFileID, NC_FLOAT, nDomainDims, LogInfo);
```

Comment:
`create_netCDF_var("domain", ...)` -- don't hard code the the domain name, it is user specified, i.e., `varNC[DOMAIN_NC]`